### PR TITLE
chore: update README to include info about error handling and tuning

### PR DESCRIPTION
**Pull request by cprice404**
Just a first stab at including some text that gives some treatment
to our error handling and tuning considerations.  This will
surely evolve over time.

### CODE REVIEW

**Comment by cprice404** [comment]
if we like this content we will need to tweak the generator to be a little more tolerant for this.

**Review by pgautier404 — DISMISSED** [review]
This sounds great to me! Now if you can just get the README validator to approve it :-)

File: README.template.md [hunk 1]
```diff
@@ -25,4 +25,41 @@ Here is a quickstart you can use in your own project:
...
+Error cases in Momento are surfaced to developers as part of the return values of the method calls, as opposed
+to by throwing exceptions.  This makes them more visible, and allows your IDE to be more helpful in ensuring that
+you've handled the ones you care about.  (For more on our philosophy about this, see our blog post on why
+[Exceptions are bugs](https://www.gomomento.com/blog/exceptions-are-bugs).  And send us any feedback you have!)
+
+The preferred way of interpreting the return values from the Momento .NET methods is using [Pattern matching](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching).  Here's a quick example:
+
+```csharp
+CacheGetResponse getResponse = await client.GetAsync(CACHE_NAME, KEY);
+if (getResponse is CacheGetResponse.Hit hitResponse)
```
Comment (eaddingtonwhite) [inline README.template.md:39]: What other response types are possible? Can we provide example of all possible cases or link out to external source?

Comment (cprice404) [reply README.template.md:39]: @eaddingtonwhite I added another commit that expands the treatment of this a bit further, lmk what you think.

Comment (cprice404) [reply README.template.md:39]: @eaddingtonwhite I'm going to merge this to make sure the README gets updated properly but I am more than happy to continue refining this part of the content if you have more input.

**Review by poppoerika — APPROVED** [review]

**Review by pgautier404 — APPROVED** [review]

File: README.template.md (hunk 1 above)
Comment (eaddingtonwhite) [reply README.template.md:39]: Looks good!

**Pull request merged**